From: Robert Rostohar <Robert.Rostohar@arm.com>
Date: Mon, 4 Jan 2021 14:26:45 +0100
Subject: [PATCH] Update doxygen project file

Use environment variables (instead of CMake variables)
which enables building documentation without CMake.
---
 doxygen/Doxyfile.in | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/doxygen/Doxyfile.in b/doxygen/Doxyfile.in
index 60b1ea4b..10eb3900 100644
--- a/doxygen/Doxyfile.in
+++ b/doxygen/Doxyfile.in
@@ -45,7 +45,7 @@ PROJECT_NAME           = "TF-M Reference Manual"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = @CMAKE_PROJECT_VERSION@
+PROJECT_NUMBER         = @DOXYCFG_TFM_VERSION@
 
 # Using the PROJECT_BRIEF tag one can provide an optional one line description
 # for a project that appears at the top of each page and should give viewer a
@@ -58,7 +58,7 @@ PROJECT_BRIEF          = Trusted Firmware-M
 # pixels and the maximum width should not exceed 200 pixels. Doxygen will copy
 # the logo to the output directory.
 
-PROJECT_LOGO           = @CMAKE_SOURCE_DIR@/doxygen/TrustedFirmware-Logo_icon.png
+PROJECT_LOGO           = @TFM_ROOT_DIR@/doxygen/TrustedFirmware-Logo_icon.png
 
 # The OUTPUT_DIRECTORY tag is used to specify the (relative or absolute) path
 # into which the generated documentation will be written. If a relative path is
@@ -169,7 +169,7 @@ FULL_PATH_NAMES        = YES
 # will be relative from the directory where doxygen is started.
 # This tag requires that the tag FULL_PATH_NAMES is set to YES.
 
-STRIP_FROM_PATH        = "@CMAKE_SOURCE_DIR@"
+STRIP_FROM_PATH        = "@TFM_ROOT_DIR@"
 
 # The STRIP_FROM_INC_PATH tag can be used to strip a user-defined part of the
 # path mentioned in the documentation of a class, which tells the reader which
@@ -178,7 +178,7 @@ STRIP_FROM_PATH        = "@CMAKE_SOURCE_DIR@"
 # specify the list of include paths that are normally passed to the compiler
 # using the -I flag.
 
-STRIP_FROM_INC_PATH    = "@CMAKE_SOURCE_DIR@"
+STRIP_FROM_INC_PATH    = "@TFM_ROOT_DIR@"
 
 # If the SHORT_NAMES tag is set to YES, doxygen will generate much shorter (but
 # less readable) file names. This can be useful is your file systems doesn't
@@ -797,10 +797,10 @@ WARN_LOGFILE           =
 # spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  = "@CMAKE_SOURCE_DIR@/doxygen" \
-                         "@CMAKE_SOURCE_DIR@/interface" \
-                         "@CMAKE_SOURCE_DIR@/secure_fw" \
-                         "@CMAKE_SOURCE_DIR@/test"
+INPUT                  = "@TFM_ROOT_DIR@/doxygen" \
+                         "@TFM_ROOT_DIR@/interface" \
+                         "@TFM_ROOT_DIR@/secure_fw" \
+                         "@TFM_ROOT_DIR@/test"
 
 
 # This tag can be used to specify the character encoding of the source files
-- 
